code/apps/Managed Software Center/Managed Software Center/Controllers/MSCAlertController.swift

```diff
@@ -161,7 +161,7 @@ class MSCAlertController: NSObject {
         } else if updatesRequireLogout() || installRequiresLogout() {
             let alert = NSAlert()
             alert.messageText = NSLocalizedString(
-                "Logout Required", comment: "Logput Required title")
```

Comment:
This should then roll down to the Localizable strings. I see it in Base.lproj/Localizable.strings and en.lproj/Localizable.strings

Reply:
I don't see any results for `grep -R "Logput" .` in the repo on this branch. Am I missing something?

Reply:
https://github.com/munki/munki/blob/Munki3dev/code/apps/Managed%20Software%20Center/Managed%20Software%20Center/Base.lproj/Localizable.strings#L107
https://github.com/munki/munki/blob/Munki3dev/code/apps/Managed%20Software%20Center/Managed%20Software%20Center/en.lproj/Localizable.strings#L107

Reply:
...and now I think I understand why grep doesn't find those instances: Xcode strings files are UTF-16 encoded text files; grep expects ASCII. Many other files are UTF-8 encoded, which is much much closer to ASCII (much code is encoded exactly the same as in an ASCII file).

Reply:
Horrible hack:
```# grep -R "L.o.g.p.u.t." .
Binary file ./code/apps/Managed Software Center/Managed Software Center/en.lproj/Localizable.strings matches
Binary file ./code/apps/Managed Software Center/Managed Software Center/Base.lproj/Localizable.strings matches```

This "works" because the ASCII character set is encoded as two bytes; the second byte is 00; but we just use '.' to match on any character. One _could_ be more specific with

```# grep -R "L\x00o\x00g\x00p\x00u\x00t\x00" .
Binary file ./code/apps/Managed Software Center/Managed Software Center/en.lproj/Localizable.strings matches
Binary file ./code/apps/Managed Software Center/Managed Software Center/Base.lproj/Localizable.strings matches```

but that's a lot harder and more tedious to type.

Reply:
Great catch. I fixed those two instances in 4256580, and I'll go back and look for other changes in the UTF-16 files now.

Reply:
Using Sublime (which I verified successfully searches binary/utf-16), I went back and searched for more instances of every correction I made. I caught and fixed a few more typos in *.py files, but no more issues were found in the strings. 👍